src/azure-cli/azure/cli/command_modules/role/_params.py

```diff
@@ -287,7 +287,7 @@ def load_arguments(self, _):
         c.argument('object_type', options_list=['--type', '-t'], help='object type filter, e.g. "application", "servicePrincipal"  "group", etc')
 
     with self.argument_context('role') as c:
-        c.argument('scope', help='scope at which the role assignment or definition applies to, e.g., /subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333, /subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceGroups/myGroup, or /subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceGroups/myGroup/providers/Microsoft.Compute/virtualMachines/myVM')
+        c.argument('scope', help='scope at which the role assignment or definition applies to, e.g., /subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333, /subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceGroups/myGroup, or /subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceGroups/myGroup/providers/Microsoft.Compute/virtualMachines/myVM without trailing slash.')
```

Comment:
**ARM's resource ID is not a URL**, so adding a trailing slash is not a valid usage. There are many other places where an ARM resource ID can be specified, like `--ids` parameter from `az vm show`:

```
> az vm show -h

Command
    az vm show : Get the details of a VM.

Resource Id Arguments
    --ids                  : One or more resource IDs (space-delimited). It should be a complete
                             resource ID containing all information of 'Resource Id' arguments. You
                             should provide either --ids or other 'Resource Id' arguments.
```

Should we include "without trailing slash" in all these help messages? 🤔

Reply:
A user may even pass `--scope /subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceGroups/myGroup/providers/Microsoft.Compute/virtualMachines/myVM without trailing slash`, interally!

Reply:
true, agreed. Let me close the PR then?

Reply:
Thanks @RakeshMohanMSFT !